kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/ZipFileRecordConverter.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ *
+ *  * Copyright 2019 The Hyve
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *   http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *  *
+ *
+ */
+
+package org.radarbase.connect.upload.converter
+
+import org.radarbase.connect.upload.api.ContentsDTO
+import org.radarbase.connect.upload.api.LogLevel
+import org.radarbase.connect.upload.api.RecordDTO
+import org.radarbase.connect.upload.exception.ProcessorNotFoundException
+import org.slf4j.LoggerFactory
+import java.io.FilterInputStream
+import java.io.IOException
+import java.io.InputStream
+import java.lang.Exception
+import java.util.zip.ZipEntry
+import java.util.zip.ZipInputStream
+
+abstract class ZipFileRecordConverter(sourceType: String, listOfDataProcessors: List<DataProcessor>) : RecordConverter(sourceType) {
+
+    private var processors: Map<String, DataProcessor> = listOfDataProcessors.map { it.schemaType to it }.toMap()
+
+    override fun processData(contents: ContentsDTO, inputStream: InputStream, record: RecordDTO, timeReceived: Double): List<TopicData> {
+        log(LogLevel.INFO,"Retrieved file content from record id ${record.id} and filename ${contents.fileName}")
+        val convertedTopicData = mutableListOf<TopicData>()
+        try {
+            val zippedInput = ZipInputStream(inputStream)
+            var zippedEntry: ZipEntry? = null
+            zippedInput.use {
+                while ({ zippedEntry = zippedInput.nextEntry; zippedEntry }() != null) {
+                    val entryName = zippedEntry!!.name.trim()
+                    logger.info("Processing entry $entryName from record ${record.id}")
+                    convertedTopicData.addAll(processContent(object : FilterInputStream(zippedInput) {
+                        @Throws(IOException::class)
+                        override fun close() {
+                            logger.info("Closing entry $entryName")
+                            zippedInput.closeEntry()
+                        }
+                    }, entryName, timeReceived))
+                }
+            }
+            convertedTopicData.last().endOfFileOffSet = true
+        } catch (exe: IOException) {
+            logger.error("Failed to process zipped input from record ${record.id}", exe)
+        } catch (exe: Exception) {
+            logger.error("Could not process record ${record.id}", exe)
```

Comment:
These messages should probably go in the full log, and be rethrown so the converter can mark the processing as failed.